### PR TITLE
feat: update module patterns

### PR DESCRIPTION
**Pull request by valerydluski**
#### 🤔 This is a ...

- [ ] 🌟 New task
- [ ] 🌐 New module
- [ ] ⚙️ Update to an existing task
- [ ] 🔧 Update to an existing module
- [ ] 🔗 Update or addition of external resources or links
- [ ] 🐛 Fix in a task or related content
- [ ] 🛠 Fix in a module or related content
- [ ] ✏️ Fixed a typo or grammatical error
- [ ] 🔗 Fixed a broken link
- [ ] ❓ Other (specify: **\*\*\*\***\_\_\_\_**\*\*\*\***)

#### Description

- **Brief Overview:**
  
- **Implementation Approach:**
  

#### Additional Information

- **Screenshots/Links:**
  
- [ ] Related Issues:

#### Checklist

- [ ] ✅ I have performed a self-review of my own code.
- [ ] 📝 I have commented my code, particularly in hard-to-understand areas.
- [ ] 🔧 I have made corresponding changes to the documentation (if applicable).
- [ ] 🚫 My changes generate no new warnings or errors.

### CODE REVIEW

File: stage2/modules/design-patterns/README.md [hunk 1]
```diff
@@ -1,38 +1,62 @@
...
-
-## Дополнительные материалы
-
-- [MVC](https://youtu.be/1Nk09Z4ca4A?t=5061)
-- [Design Patterns](https://www.youtube.com/watch?v=pH7LauSInc8)
+# [Stage#2.](../../) Design Patterns 🏗️
+
+## Module Overview 📚
+
+The "Design Patterns" module delves into the foundational principles of software design patterns, categorizing them into creational, structural, and behavioral patterns. This course aims to equip students with the knowledge to apply these patterns in various programming scenarios, enhancing code reusability, scalability, and maintainability.
```
Comment (haslie22) [inline stage2/modules/design-patterns/README.md:5]:
```suggestion
The "Design Patterns" module delves into the foundational principles of software design patterns, categorizing them into creational, structural, and behavioral patterns. This module aims to equip students with the knowledge to apply these patterns in various programming scenarios, enhancing code reusability, scalability, and maintainability.
```

**Review by haslie22 — APPROVED** [review]
LGTM 👍

File: stage2/modules/design-patterns/README.md [hunk 2]
```diff
@@ -1,38 +1,62 @@
...
+- **6 hours**
+
+## Theory 📖
+
+Students are encouraged to explore the following resources for a comprehensive understanding:
+
+1. **Introduction to Design Patterns:**
+
+   - [Application architecture and patterns](https://doka.guide/tools/architecture-and-design-patterns/)
+   - [MVC](https://doka.guide/tools/architecture-mvc/)
```
Comment (KristiBo) [inline stage2/modules/design-patterns/README.md:28]: These materials are in Russian, perhaps we need to write it?

**Review by KristiBo — APPROVED** [review]
Good job :+1:

**Review by stardustmeg — APPROVED** [review]

**Pull request merged**